extensions/warp-ipfs/src/lib.rs

```diff
@@ -1355,158 +1354,36 @@ impl IdentityInformation for WarpIpfs {
-            .await
-    }
-
-    async fn message_status(
-        &self,
-        conversation_id: Uuid,
-        message_id: Uuid,
-    ) -> Result<MessageStatus, Error> {
-        self.messaging_store()?
-            .message_status(conversation_id, message_id)
-            .await
+            .await?;
+        store.get_conversation(id).await
     }
 
-    async fn get_messages(
+    async fn get_conversation(
         &self,
         conversation_id: Uuid,
-        opt: MessageOptions,
-    ) -> Result<Messages, Error> {
-        self.messaging_store()?
-            .get_messages(conversation_id, opt)
-            .await
-    }
-
-    async fn send(&mut self, conversation_id: Uuid, value: Vec<String>) -> Result<(), Error> {
-        self.messaging_store()?
-            .send_message(conversation_id, value)
-            .await
-    }
-
-    async fn edit(
-        &mut self,
-        conversation_id: Uuid,
-        message_id: Uuid,
-        value: Vec<String>,
-    ) -> Result<(), Error> {
+    ) -> Result<Box<dyn Conversation>, Error> {
         self.messaging_store()?
-            .edit_message(conversation_id, message_id, value)
-            .await
-    }
-
-    async fn delete(
-        &mut self,
-        conversation_id: Uuid,
-        message_id: Option<Uuid>,
-    ) -> Result<(), Error> {
-        let store = self.messaging_store()?;
-        match message_id {
-            Some(id) => store.delete_message(conversation_id, id).await,
-            None => store.delete_conversation(conversation_id).await.map(|_| ()),
-        }
-    }
-
-    async fn react(
-        &mut self,
-        conversation_id: Uuid,
-        message_id: Uuid,
-        state: ReactionState,
-        emoji: String,
-    ) -> Result<(), Error> {
-        self.messaging_store()?
-            .react(conversation_id, message_id, state, emoji)
-            .await
-    }
-
-    async fn pin(
-        &mut self,
-        conversation_id: Uuid,
-        message_id: Uuid,
-        state: PinState,
-    ) -> Result<(), Error> {
-        self.messaging_store()?
-            .pin_message(conversation_id, message_id, state)
-            .await
-    }
-
-    async fn reply(
-        &mut self,
-        conversation_id: Uuid,
-        message_id: Uuid,
-        value: Vec<String>,
-    ) -> Result<(), Error> {
-        self.messaging_store()?
-            .reply(conversation_id, message_id, value)
+            .get_conversation(conversation_id)
             .await
     }
 
-    async fn embeds(&mut self, _: Uuid, _: Uuid, _: EmbedState) -> Result<(), Error> {
-        Err(Error::Unimplemented)
-    }
-
-    async fn update_conversation_settings(
-        &mut self,
-        conversation_id: Uuid,
-        settings: ConversationSettings,
-    ) -> Result<(), Error> {
-        self.messaging_store()?
-            .update_conversation_settings(conversation_id, settings)
-            .await
+    async fn list_conversations(&self) -> Result<Vec<Box<dyn Conversation>>, Error> {
```

Comment:
can't we just return a `&[Box<..>]` here or `impl Iterator<Item = Box<dyn Conversation>> + '_` here? 🤔

Reply:
If I recall doing that would cause the trait not to object safe

Reply:
I think there is a different issue with doing that: [Send and Sync](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=90dd7c07c6d58423f71cff1afd84ff44).

Reply:
> Do we need object safety WarpIpfs trait?
It is preferred at this point to make sure there is object safety.